### DISCUSSION TITLE
Some way to distinguish WLA symbol maps from other symbol maps.

### DISCUSSION THREAD

**Discussion by Pokeman2003** [vhelin/wla-dx#612] [Ideas]
I quite like this assembler, but the Gearsystem is not compatible with its symbol maps. The developer seems interested in adding support for the symbol maps that WLA-Linker generates, but I've known for a bit that there's really no way to distinguish a WLA symbol map from any other relatively similar symbol map. Most notably, the `[source files v2]` and `[definitions]` sections seem to really trip up Gearsystem's parser for being so similar to another symbol map's format. It stands to reason then that any parsers which DO support multiple formats may have trouble figuring out which format you're trying to feed them.

I think this confusion for multi-format parsers could easily be solved with some distinguishing factor within the symbol maps that could be immediately parsed and recognized as "yup, that's WLA," regardless of who generated the symbol map(so no relying on the `this file was created with wlalink` comment generated at the top.) I don't really have a fantastic solution, but I've got a couple of spitball ideas. Perhaps linkers could generate an intentionally blank section that's just called `[wlasymbol]`? Or maybe something can fall under `[information]`, perhaps a line that just says `wlasymbol true`?

**Comment by vhelin** [comment]
Haven't looked what would be the best solution, how about under [information] something like "linker = wlalink" and "linkerversion = vx.y"? I'll come back to this tomorrow evening and implement something like that...

**Comment by vhelin** [comment]
Looking at this issue. I can see WLALINK already writes a commet like header to the symbol file that contains the string "wlalink" - that could be used to identify that the symbol file was written by WLALINK. But I'll add "format wlasymbolfile" to [information] if that helps...

**Reply by Pokeman2003** [reply]
My main concern about the comment is that the comment could be altered if it's made by a different linker that supports the same symbol map or was stripped by another system to save a few bytes on the hard disk.
`regardless of who generated the symbol map(so no relying on the this file was created with wlalink comment generated at the top.) `

**Comment by vhelin** [comment]
Or actually I know nothing of other symbol files so I'll follow your advice here and add "wlasymbol true" under [information]...

**Comment by vhelin** [comment]
Ok, check out the latest sources

**Comment by vhelin** [comment]
@Pokeman2003 Please let me know if the latest sources solve your issue. I'm planning on releasing the next release quite soon so having this fixed before the release is quite important. :)

**Reply by Pokeman2003** [reply]
As far as I can tell, this fixes my "issue!" Thank you for this implementation.
Apologies for not getting back sooner, I've been a bit busy recently.
